lib/model-builder.js

```diff
@@ -168,6 +169,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
     return ModelBaseClass.extend(className, properties, settings);
   }
 
+  // initializing base model inheritance rank if not set already
+  ModelBaseClass.__rank = ModelBaseClass.__rank || 1;
```

Comment:
Recalculating ranks is not a bad idea. We probably will need to have `mixins` which don't have an explicit chain.